vendors/st/boards/stm32l475_discovery/ports/common_io/iot_uart.c

```diff
@@ -0,0 +1,689 @@
+            .Parity         = UART_PARITY_NONE,
+            .Mode           = UART_MODE_TX_RX,
+            .HwFlowCtl      = UART_HWCONTROL_NONE,
+            .OverSampling   = UART_OVERSAMPLING_16,
+            .OneBitSampling = UART_ONE_BIT_SAMPLE_DISABLE
+        },
+        .AdvancedInit =
+        {
+            .AdvFeatureInit = UART_ADVFEATURE_NO_INIT
+        },
+        .pTxBuffPtr = NULL,
+        .TxXferSize = 0,
+        .TxXferCount = 0,
+        .pRxBuffPtr = NULL,
+        .RxXferSize = 0,
+        .RxXferCount = 0
+    },
+};
+
+/**
+ * @brief Statically initialized map of STM's HALContext.
+ */
+static const STM32_UART_HalContext_t xUartMap[] =
+{
+    {
+        .pxHuart = &xUartHandleMap[ 0 ],
+        .eIrqNum = USART1_IRQn,
+    },
+    {
+        .pxHuart = &xUartHandleMap[ 1 ],
+        .eIrqNum = USART2_IRQn,
+    },
+    {
+        .pxHuart = &xUartHandleMap[ 2 ],
+        .eIrqNum = USART3_IRQn,
+    },
+    {
+        .pxHuart = &xUartHandleMap[ 3 ],
+        .eIrqNum = UART4_IRQn,
+    },
+    {
+        .pxHuart = &xUartHandleMap[ 4 ],
+        .eIrqNum = UART5_IRQn,
+    },
+};
+
+static IotUARTDescriptor_t xUart0 =
+{
+    .pvUserCallbackContext = NULL,
+    .pvHalContext          = ( STM32_UART_HalContext_t * ) &xUartMap[ 0 ],
+    .xUartCallback         = NULL,
+    .xSemphr               = NULL,
+    .sOpened               = IOT_UART_CLOSED,
+};
+
+static IotUARTDescriptor_t xUart1 =
+{
+    .pvUserCallbackContext = NULL,
+    .pvHalContext          = ( STM32_UART_HalContext_t * ) &xUartMap[ 1 ],
+    .xUartCallback         = NULL,
+    .xSemphr               = NULL,
+    .sOpened               = IOT_UART_CLOSED,
+};
+
+static IotUARTDescriptor_t xUart2 =
+{
+    .pvUserCallbackContext = NULL,
+    .pvHalContext          = ( STM32_UART_HalContext_t * ) &xUartMap[ 2 ],
+    .xUartCallback         = NULL,
+    .xSemphr               = NULL,
+    .sOpened               = IOT_UART_CLOSED,
+};
+
+static IotUARTDescriptor_t xUart3 =
+{
+    .pvUserCallbackContext = NULL,
+    .pvHalContext          = ( STM32_UART_HalContext_t * ) &xUartMap[ 3 ],
+    .xUartCallback         = NULL,
+    .xSemphr               = NULL,
+    .sOpened               = IOT_UART_CLOSED,
+};
+
+static IotUARTDescriptor_t xUart4 =
+{
+    .pvUserCallbackContext = NULL,
+    .pvHalContext          = ( STM32_UART_HalContext_t * ) &xUartMap[ 4 ],
+    .xUartCallback         = NULL,
+    .xSemphr               = NULL,
+    .sOpened               = IOT_UART_CLOSED,
+};
+
+static IotUARTHandle_t const pxUarts[] = { &xUart0, &xUart1, &xUart2, &xUart3, &xUart4 };
+
+IotUARTHandle_t iot_uart_open( int32_t lUartInstance )
+{
+    IotUARTHandle_t xHandle = NULL;
+    UART_HandleTypeDef * xHandleHAL = NULL;
+
+    if( ( lUartInstance < IOT_USART_PORTS ) && ( lUartInstance >= 0 ) )
+    {
+        xHandle = pxUarts[ lUartInstance ];
+
+        /*lint !e9079 !e9087 handle passed should be of UART_HandleTypeDef *. */
```

Comment:
Looks like xHandle needs to be of type IoTUARTHandle_t.

Reply:
This comment is confusing. Let me remove it.